src/darwin/Framework/CHIP/MTRDeviceControllerStorageDelegate.h

```diff
@@ -0,0 +1,119 @@
+/**
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRDeviceController.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, MTRStorageSecurityLevel) {
+    // Data must be stored in secure (encrypted) storage.
+    MTRStorageSecurityLevelSecure,
+    // Data may be stored in the clear.
+    MTRStorageSecurityLevelNotSecure,
+} MTR_NEWLY_AVAILABLE;
+
+typedef NS_ENUM(NSUInteger, MTRStorageSharingType) {
+    // Data must not be shared at all (just store locally).
+    MTRStorageSharingTypeNotShared,
+
+    // Data may be shared, but only between controllers that have the same node
```

Comment:
Same as below, but I think we should be explicit, and not leave any room for interpretation of what to do with this. For example: When folks go to adopt this they shouldn't have to ask you what this means, they should know the sync strategy and domain from the description.